src/electionguard/decryption.py

```diff
@@ -463,15 +473,27 @@ def compensate_decrypt(
     :param missing_guardian_backup: Missing guardians backup
     :param ciphertext: the `ElGamalCiphertext` that will be partially decrypted
     :param extended_base_hash: the extended base hash of the election that
-                                was used to generate t he ElGamal Ciphertext
+                                was used to generate the ElGamal Ciphertext
     :param nonce_seed: an optional value used to generate the `ChaumPedersenProof`
                         if no value is provided, a random number will be used.
     :return: a `Tuple[ElementModP, ChaumPedersenProof]` of the decryption and its proof
     """
     if nonce_seed is None:
         nonce_seed = rand_q()
 
-    partial_secret_key = missing_guardian_backup.coordinate
+    encryption_seed = get_backup_seed(
+        guardian_keys.owner_id,
+        guardian_keys.sequence_order,
+    )
+    bytes_optional = missing_guardian_backup.encrypted_coordinate.decrypt(
+        guardian_keys.key_pair.secret_key, encryption_seed
+    )
+    if bytes_optional is None:
+        return None
```

Comment:
I do not like propagating this "if null return null pattern", but this PR did not seem like the right place to start fixing it, especially since the source of this problem is in the otherwise untouched hashed elgamal section of code.

Reply:
I agree, but I have no qualms about throwing an error if you wish.